src/sentinel.rs

```diff
@@ -56,237 +56,229 @@ pub struct Sentinel<'a> {
 
 impl<'a> Sentinel<'a> {
   pub fn new(send_get_keys: &'a mut SendGetKeys) -> Sentinel<'a> {
-  	Sentinel {
-  	  send_get_keys_: send_get_keys,
-  	  node_accumulator_: NodeAccumulatorType::new(20),
-  	  group_accumulator_: NodeAccumulatorType::new(20),
-  	  group_key_accumulator_: KeyAccumulatorType::new(20),
-  	  node_key_accumulator_: KeyAccumulatorType::new(20)
-  	}
+    Sentinel {
+      send_get_keys_: send_get_keys,
+      node_accumulator_: NodeAccumulatorType::new(20),
+      group_accumulator_: NodeAccumulatorType::new(20),
+      group_key_accumulator_: KeyAccumulatorType::new(20),
+      node_key_accumulator_: KeyAccumulatorType::new(20)
+    }
   }
 
-  // pub fn get_send_get_keys(&'a mut self) -> &'a mut SendGetKeys { self.send_get_keys }
-
   pub fn add(&mut self, header : message_header::MessageHeader, type_tag : types::MessageTypeTag,
-  	         message : types::SerialisedMessage) -> Option<ResultType> {
-  	match type_tag {
-  		types::MessageTypeTag::GetClientKeyResponse => {
-  			if header.is_from_group() {
-	  			let keys = self.node_key_accumulator_.add(header.from_group().unwrap(),
-			  				                                    (header.clone(), type_tag, message),
-			  				                                    header.from_node());
-		      if keys.is_some() {
-		      	let key = (header.from_group().unwrap(), header.message_id());
-		      	let messages = self.node_accumulator_.get(&key);
-		      	if messages.is_some() {
-	      			let resolved = self.resolve(self.validate_node(messages.unwrap().1,
-	      				                                             keys.unwrap().1), false);
-	      			if resolved.is_some() {
-      					self.node_accumulator_.delete(key);
-      					return resolved;
-	      			}
-		      	}
-		      }
-			  }
-  		}
-			types::MessageTypeTag::GetGroupKeyResponse => {
-				if header.is_from_group() {
-	  			let keys = self.group_key_accumulator_.add(header.from_group().unwrap(),
-			  				                                     (header.clone(), type_tag, message),
-			  				                                     header.from_node());
-	  			if keys.is_some() {
-		      	let key = (header.from_group().unwrap(), header.message_id());
-		      	let messages = self.group_accumulator_.get(&key);
-		      	if messages.is_some() {
-	      			let resolved = self.resolve(self.validate_group(messages.unwrap().1,
-	      				                                              keys.unwrap().1), true);
-	      			if resolved.is_some() {
-      					self.group_accumulator_.delete(key);
-      					return resolved;
-	      			}
-		      	}
-			    }
-			  }
-			}
-			_ => {
-				if header.is_from_group() {
-					let key = (header.from_group().unwrap(), header.message_id());
-					if !self.group_accumulator_.have_name(&key) {
-						//self.key_getter_traits_.get_group_key(header.from_group().unwrap());
-						self.send_get_keys_.get_group_key(header.from_group().unwrap());
-					} else {
```

Comment:
this if-statement is only scoped over `self.send_get_keys_.get_group_key(header.from_group().unwrap());`